matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/migration/MigrateSessionTo030.kt

```diff
@@ -35,16 +36,23 @@ internal class MigrateSessionTo030(realm: DynamicRealm) : RealmMigrator(realm, 3
                 .equalTo(ChunkEntityFields.IS_LAST_FORWARD, false)
                 .findAll()
 
+        var nbOfDeletedObjects = 0
         chunks.forEach { chunk ->
-            chunk.getList(ChunkEntityFields.TIMELINE_EVENTS.`$`).clearWith { timelineEvent ->
+            val realmObjectToDelete = mutableListOf<DynamicRealmObject>()
+            chunk.getList(ChunkEntityFields.TIMELINE_EVENTS.`$`).forEach { timelineEvent ->
                 // Don't delete state events
                 val event = timelineEvent.getObject(TimelineEventEntityFields.ROOT.`$`)
                 if (event?.isNull(EventEntityFields.STATE_KEY) == true) {
-                    event.deleteFromRealm()
-                    timelineEvent.deleteFromRealm()
+                    realmObjectToDelete.add(event)
+                    realmObjectToDelete.add(timelineEvent)
```

Comment:
TimelineEvent can always be delete, it's just the root Event which should not be kept if state event.

Reply:
Good to know, but this is not new? So the `clearWith` may work if we always delete the TimelineEvent.

Reply:
OK, I will test that